bffnt.py

```diff
@@ -503,32 +522,38 @@ def save(self, filename):
 
         file_.seek(section_count_pos)
         file_.write(struct.pack('%sI' % self.order, section_count))
+        if self.verbose:
+            print('Done!')
 
     def _parse_header(self, data):
-        magic, bom, header_size, version, file_size, sections = struct.unpack(FFNT_HEADER_STRUCT, data)
-
-        if magic != FFNT_HEADER_MAGIC:
-            print('Invalid FFNT magic bytes: %s (expected %s)' % (magic, FFNT_HEADER_MAGIC))
-            self.invalid = True
-            return
-
+        bom = struct.unpack_from('>H', data, 4)[0]
         if bom == 0xFFFE:
-            self.order = '>'
-        elif bom == 0xFEFF:
             self.order = '<'
+        elif bom == 0xFEFF:
+            self.order = '>'
 
         if self.order is None:
             print('Invalid byte-order marker: 0x%x (expected 0xFFFE or 0xFEFF)' % bom)
             self.invalid = True
             return
+            
+        magic, bom, header_size, self.version, file_size, sections, unknown = struct.unpack(FFNT_HEADER_STRUCT % self.order, data)
 
-        if version != VERSION:
-            print('Unknown version: 0x%08x (expected 0x%08x)' % (version, VERSION))
+        if magic not in FFNT_HEADER_MAGIC:
+            print('Invalid FFNT magic bytes: %s (expected %s)' % (magic, FFNT_HEADER_MAGIC))
+            self.invalid = True
+            return
+        self.filetype = magic.decode('ascii').lower()
+
+
+        if self.version not in VERSIONS:
+            print('Unknown version: 0x%08x (expected 0x%08x or 0x%08x)' % (self.version, 0x03000000, 0x04000000))
             self.invalid = True
             return
 
         if header_size != FFNT_HEADER_SIZE:
             print('Invalid header size: %d (expected %d)' % (header_size, FFNT_HEADER_SIZE))
+            blinh
```

Comment:
Conflict may be because you were developing on an old branch that we already merged into `master`